cpp/include/cuvs/neighbors/ivf_flat_helpers.cuh

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2023-2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
```

Comment:
We can't expose `cuh` files through the public API. Everything needs to be compiled and accessible from host- this file should be moved to `src/neighbors/ivf_flat`, a source file called `ivf_flat_helpers.cu` created that instantiates everything in this file, and a header file exposed in `include/cuvs/neighbors/ivf_flat_helpers.hpp` that declares the function prototypes.